src/aead/gcm.rs

```diff
@@ -250,16 +253,16 @@ struct u128 {
 const HTABLE_LEN: usize = 16;
 
 #[repr(transparent)]
-pub struct Xi(Block);
+pub struct Xi([u8; BLOCK_LEN]);
```

Comment:
It seems like we need types like this?:

```
struct H([BigEndian<u64>; 2);
struct Xi([BigEndian<u64>; 2);
```

or maybe:

```
struct H([u64; 2);
struct Xi([u64; 2);
```

depending on what the actual endianness is supposed to be.

In particular, doesn't the assembly code require these to have the same ABI (alignment, etc.) as u64?

Reply:
I didn't dig into the assembly, and I'm sure you're right. Will take a closer look on the next round of refactors.